openedx/features/subscriptions/utils.py

```diff
@@ -27,6 +27,9 @@ def is_course_accessible_with_subscription(user, course):
     """
     Check if user has access to a course enrolled through subscription.
     """
+    if user.is_anonymous:
+        return ACCESS_GRANTED
```

Comment:
`ACCESS_DENIED`

Reply:
done

Reply:
we need to determine which user can access the course. there is a difference between accessing course and accessing course content. we only want to restrict access to a course if a user had enrolled in it using a subscription. So that means an anonymous user is allowed to access the course, but not the course content which is handled elsewhere.

Reply:
Thanks for the explanation